src/barril/units/_scalar.py

```diff
@@ -238,23 +238,34 @@ def GetFormatted(self, unit=None, value_format=None):
         )
 
     # Compare --------------------------------------------------------------------------------------
+    def _GetValueInDefaultUnit(self):
+        try:
+            return self.GetValue(
+                self._unit_database.GetDefaultUnit(self._quantity._category)
+            )
+        except InvalidQuantityTypeError:
+            return self._value
 
     def __eq__(self, other):
         return (
             type(self) is type(other)
-            and self._value == other.value
-            and self._quantity == other._quantity
+            and self._quantity._category == other._quantity._category
+            and self._GetValueInDefaultUnit() == other._GetValueInDefaultUnit()
```

Comment:
You are using `_GetValueInDefaultUnit` here because of `__hash__` requirements correct?

If so, codify this in a test:

```python
def test_hashing():
   m = Scalar(1, 'm')
   cm = Scalar(100, 'cm')
   assert m == cm
   assert hash(m) == hash(cm)
```

Reply:
Two different objects having the same hash is ok.
This is being used to avoid `(a == b) != (b == a)`.

Reply:
Hmm yes, but my comment and suggestion still stands, I think

Reply:
> You are using _GetValueInDefaultUnit here because of __hash__ requirements correct?
Yes